### DISCUSSION TITLE
Leaves Auto Incrementation

### DISCUSSION THREAD

**Discussion by sharukhkhan234** [horilla-opensource/horilla#202] [General]
Hi!
We want to add some features regarding the leaves. for instance, if we want to increment 2 earned leaves each month automatically, manually we can add leaves in each month, but therefore, to add features that increment auto 2 leaves every month, it will be good if these feature sets for all employee types. and Secondly, employees or users can only request 2 leaves a month.

**Comment by horilla-opensource** [comment]
> Hi! We want to add some features regarding the leaves. for instance, if we want to increment 2 earned leaves each month automatically, manually we can add leaves in each month, but therefore, to add features that increment auto 2 leaves every month, it will be good if these feature sets for all employee types. and Secondly, employees or users can only request 2 leaves a month.



> Hi! We want to add some features regarding the leaves. for instance, if we want to increment 2 earned leaves each month automatically, manually we can add leaves in each month, but therefore, to add features that increment auto 2 leaves every month, it will be good if these feature sets for all employee types. and Secondly, employees or users can only request 2 leaves a month.

Hi @sharukhkhan234 ,
You have the option to reset the leave days which will increment the leave days monthly.
When you create a leave type, set the reset period as monthly and total days as 2.
Please refer the below screenshot.
<img width="639" alt="Screenshot 2024-06-05 at 12 24 10 PM" src="https://github.com/horilla-opensource/horilla/assets/131998600/a2c2818c-82fb-4f2c-8365-c6ab4c9ce0a6">

**Reply by sharukhkhan234** [reply]
Needs of few improvements!
No Option to choose the record of the Selected Department, on the dashboard of work records we want to select the employees based on their Departments, the conflict is all the employees showing on the work records dashboard, but it's hard to identify.

![Work Records](https://github.com/horilla-opensource/horilla/assets/157000111/e24cc978-1dae-428d-9fc1-2727add23d73)

**Reply by sharukhkhan234** [reply]
Secondly wants  "L"(leave) instead of " A"(absent) on the work records dashboard.

![Leave](https://github.com/horilla-opensource/horilla/assets/157000111/a2e540b2-c0b5-4ebc-bcd8-52cb51bf61e1)

**Reply by horilla-opensource** [reply]
You can modify that from the code only.

**Reply by horilla-opensource** [reply]
> Needs of few improvements! No Option to choose the record of the Selected Department, on the dashboard of work records we want to select the employees based on their Departments, the conflict is all the employees showing on the work records dashboard, but it's hard to identify.
> 
> ![Work Records](https://private-user-images.githubusercontent.com/157000111/336776251-e24cc978-1dae-428d-9fc1-2727add23d73.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NzMxMjUsIm5iZiI6MTcxNzY3MjgyNSwicGF0aCI6Ii8xNTcwMDAxMTEvMzM2Nzc2MjUxLWUyNGNjOTc4LTFkYWUtNDI4ZC05ZmMxLTI3MjdhZGQyM2Q3My5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwNlQxMTIwMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ODA4YWM3M2NhOWU2M2U1Y2RiZGEwZDkyMjA4NzJkMzk0NjNjZmMxYjNlNGVmNzAwY2YwOTA3YTg2YzEyM2NmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.c1Oi0B0GNQGVYuvToNJ36W3kkuLVgUXp7HBuAsYPnV0)

Will add that option in the future commits

**Comment by sharukhkhan234** [comment]
Hi!
I have added these two fields in the employee form, but I just added ( personal_info.html and models.py), can you share the employee form linking, I mean say which files are linked with the employee form? Secondly, if we want to add some fields to this project app, which files will be linked to each other?
![id-card](https://github.com/horilla-opensource/horilla/assets/157000111/58aa0bd3-a981-4568-a7d2-ea2e68cd6468)
